src/_common/events/subscription.ts

```diff
@@ -0,0 +1,26 @@
+export type UnsubscribeFunc = () => void;
+
+export class EventSubscription {
+	private unsubFunc: UnsubscribeFunc | null;
+
+	constructor(private subscriber: () => UnsubscribeFunc) {
+		this.unsubFunc = null;
+	}
+
+	subscribe() {
+		if (!this.unsubFunc) {
+			this.unsubFunc = this.subscriber();
+		}
+
+		return this;
+	}
+
+	unsubscribe() {
+		if (this.unsubFunc) {
```

Comment:
Does this mean the unsubFunc is state for whether we are subscribed or not? I think we should have an additional value like isSubscribed or something.

Reply:
what?